### PR TITLE
fix: Improve progress output timing

### LINKED ISSUE #4: Long delay for first output
The output showing the command and configuration is printed using `print!()` before the command is executed.  However, on systems which are line-buffered, that output is not actually displayed until after a newline is printed -- which happens here when the results of the command are retrieved and the `OK` or `Fail` message is printed.  For projects that take several minutes to compile, this means that the operator gets no output until compilation is complete and the command is executed.

See [the rust documentation for `std::print`](https://doc.rust-lang.org/std/macro.print.html) for a discussion of this.

This could be fixed / improved by adding a call to `io::stdout().flush()` immediately after the first part of the status message is printed (before the command is executed) so that the output is immediately displayed.

I can fix this myself, if a PR would be accepted.  Let me know.

### PR DESCRIPTION
**Pull request by mdharm**
The output showing the command and configuration is printed using print!() before the command is executed. However, on systems which are line-buffered, that output is not actually displayed until after a newline is printed -- which happens here when the results of the command are retrieved and the OK or Fail message is printed.

This commit adds a call to std::io::stdout::flush() to make the first part of the output (showing command and configuration) actually get printed to the display immediately, without waiting for the command to complete.

This does not change the content or formatting of the output; only the timing of the output is changed.  This closes #4

### CODE REVIEW

**Comment by Person-93** [comment]
I know that it's not documented anywhere but I am trying to follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) in this project. It requires that every commit have a type. Can you amend your commit and add a type? The allowed types are listed [here](https://github.com/Person-93/cargo-feature-matrix/blob/a181119bbb777589179b149c62871b9b1a60ce46/.commitlintrc.json#L12). I think "fix" is the appropriate type.

In short, please amend your commit and add "fix: " to the start of the commit message and then force push.

**Comment by mdharm** [comment]
> In short, please amend your commit and add "fix: " to the start of the commit message and then force push.

Commit updated and force-pushed.  Commitlint now passes.

May I also suggest that you release a new version to crates.io once this is merged?

**Comment by Person-93** [comment]
@mdharm thanks again! I've released a new version on [crates.io](https://crates.io/crates/cargo-feature-matrix/0.3.0) with your changes.
